### ISSUE TITLE
Tensorboard reports final epoch validation loss, *not* minimum validation loss over history

### ISSUE THREAD

**Issue by brethvoice** [tensorflow/tensorflow#45384]
**System information**
- TensorFlow version:  2.3.0
- Are you willing to contribute it:  Yes, with guidance/help to know where to look

**Describe the feature and the current behavior/state.**
Feature would change current behavior of Tensorboard.  Current behavior is that Tensorboard displays the validation loss from the final epoch of training, which is not useful when comparing different models to each other.

**Will this change the current api? How?**
This will not change the current api.

**Who will benefit with this feature?**
Everyone who uses Tensorboard to compare models to each other.

**Any Other info.**
I am currently using a standard `tf.keras.callbacks.Tensorboard` instance, sub-classed with the following method as the only modification:
```
    def on_train_end(self, logs=None):
        if os.path.exists(os.path.join(self.log_dir, "train", "plugins")):
            shutil.rmtree(os.path.join(self.log_dir, "train", "plugins"))
```

**Comment by rmothukuru** [comment]
@brethvoice,
Can you please share the code corresponding to the statement, 

> Current behavior is that Tensorboard displays the validation loss from the final epoch of training. 

so that it helps us to reproduce the issue better and understand exactly what is missing.

Thanks!

**Comment by brethvoice** [comment]
> @brethvoice,
> Can you please share the code corresponding to the statement,
> 
> > Current behavior is that Tensorboard displays the validation loss from the final epoch of training.
> 
> so that it helps us to reproduce the issue better and understand exactly what is missing.
> 
> Thanks!

The code is nearly 300 lines, so I posted it to my only Github [repo](https://github.com/brethvoice/optuna_demo_MNIST/blob/main/tensorboard_example_code.py).  It can be run from Windows 10 using local GPU if available.

**Comment by psybuzz** [comment]
Thanks for this feature request.  It seems reasonable to want to see the "best" (maximum or minimum) value of a particular metric in the table UI at a glance, rather than just the value from the final epoch/step.

If I understand how the Hparams dashboard works, the metric names are linked with series line charts shown in the Scalar dashboard.  If that is the case, and the scalar + hparam data are already plumbed to TensorBoard today, I think this might not need any API related changes.  As such, this might only require frontend fixes.
